pkg/ctl/cmdutils/configfile.go

```diff
@@ -338,6 +343,23 @@ func NewCreateClusterLoader(cmd *Cmd, ngFilter *filter.NodeGroupFilter, ng *api.
 	return l
 }
 
+func validateZonesAndNodeZones(cmd *cobra.Command) error {
+	if nodeZonesFlag := cmd.Flag("node-zones"); nodeZonesFlag != nil && nodeZonesFlag.Changed {
+		zonesFlag := cmd.Flag("zones")
+		if zonesFlag == nil || !zonesFlag.Changed {
```

Comment:
```suggestion
		if zonesFlag := cmd.Flag("zones"); zonesFlag == nil || !zonesFlag.Changed {
```
to mirror the line above 😌

Reply:
fine. :D :D

Reply:
Oh, but that actually makes this error

```go
			return errors.New("zones must be defined if node-zones is provided and must be a superset of node-zones")
```

Having to put on an `else` which is ugly. :D Sorry :))

Reply:
I don't see why it would need an `else`, wouldn't it just continue if it doesn't meet the conditions or return the error if it does (what we want)? what am I missing here? 😳

Reply:
How do you mean? If I don't put an else in there, I would have to return nil if there is no error after the inside validation finished. Something like this:

```go
func validateZonesAndNodeZones(cmd *cobra.Command) error {
	if nodeZonesFlag := cmd.Flag("node-zones"); nodeZonesFlag != nil && nodeZonesFlag.Changed {
		if zonesFlag := cmd.Flag("zones"); zonesFlag != nil && zonesFlag.Changed {
			nodeZones := strings.Split(strings.Trim(nodeZonesFlag.Value.String(), "[]"), ",")
			zones := strings.Split(strings.Trim(zonesFlag.Value.String(), "[]"), ",")
			for _, zone := range nodeZones {
				if !utilstrings.Contains(zones, zone) {
					return fmt.Errorf("node-zones %s must be a subset of zones %s; %q was not found in zones", nodeZones, zones, zone)
				}
			}
			return nil
		}
		return errors.New("zones must be defined if node-zones is provided and must be a superset of node-zones")
	}
	return nil
}
```

Otherwise it would always just fall though to the error even if the inside validation finished successfully. Or, I put it on an else like this: 

```go

func validateZonesAndNodeZones(cmd *cobra.Command) error {
	if nodeZonesFlag := cmd.Flag("node-zones"); nodeZonesFlag != nil && nodeZonesFlag.Changed {
		if zonesFlag := cmd.Flag("zones"); zonesFlag != nil && zonesFlag.Changed {
			nodeZones := strings.Split(strings.Trim(nodeZonesFlag.Value.String(), "[]"), ",")
			zones := strings.Split(strings.Trim(zonesFlag.Value.String(), "[]"), ",")
			for _, zone := range nodeZones {
				if !utilstrings.Contains(zones, zone) {
					return fmt.Errorf("node-zones %s must be a subset of zones %s; %q was not found in zones", nodeZones, zones, zone)
				}
			}
		} else if zonesFlag == nil || !zonesFlag.Changed {
			return errors.New("zones must be defined if node-zones is provided and must be a superset of node-zones")
		}
	}
	return nil
}
```

Which is, of course, way more complex to read. :)

So while the first one is fine, I don't really like the double return nil. :D But if you insist I can do that. :D

Reply:
The reason I like this approach more is that it requires less parsing in the brain. You immediately see the error cases and when it returns and why. :)